src/libraries/System.Linq/src/System/Linq/OrderBy.cs

```diff
@@ -7,12 +7,126 @@ namespace System.Linq
 {
     public static partial class Enumerable
     {
+        /// <summary>
+        /// Sorts the elements of a sequence in ascending order.
+        /// </summary>
+        /// <typeparam name="T">The type of the elements of <paramref name="source"/>.</typeparam>
+        /// <param name="source">A sequence of values to order.</param>
+        /// <returns>An <see cref="IOrderedEnumerable{TElement}"/> whose elements are sorted.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is <see langword="null"/>.</exception>
+        /// <remarks>
+        /// This method is implemented by using deferred execution. The immediate return value is an object
+        /// that stores all the information that is required to perform the action.
+        /// The query represented by this method is not executed until the object is enumerated by calling
+        /// its <see cref="IEnumerable{T}.GetEnumerator"/> method.
+        ///
+        /// Two methods are defined to extend the type <see cref="IOrderedEnumerable{TElement}"/>, which is the return type of this method.
+        /// These two methods, namely <see cref="ThenBy{TSource, TKey}(IOrderedEnumerable{TSource}, Func{TSource, TKey})"/>
+        /// and <see cref="ThenByDescending{TSource, TKey}(IOrderedEnumerable{TSource}, Func{TSource, TKey})"/>, enable you to specify additional
```

Comment:
Even though this method returns IOrderedEnumerable, I would de-emphasize chaining with `ThenBy` in the docs since it is unlikely you would want to chain with another key if comparing the element values with themselves.